community/kernel/src/test/java/org/neo4j/kernel/impl/store/SchemaStorageTest.java

```diff
@@ -115,6 +119,109 @@ public void shouldReturnIndexRuleForLabelAndProperty()
     }
 
     @Test
+    public void shouldReturnIndexRuleForLabelAndPropertyComposite() throws Exception
+    {
+        String a = "a", b = "b", c = "c", d = "d", e = "e", f = "f";
+        createSchema( db ->
+        {
+            db.schema().indexFor( Label.label( LABEL1 ) )
+              .on( a ).on( b ).on( c ).on( d ).on( e ).on( f ).create();
+        } );
+
+        IndexRule rule = storage.indexGetForSchema( NewIndexDescriptorFactory.forLabel(
+                labelId( LABEL1 ), propId( a ), propId( b ), propId( c ), propId( d ), propId( e ), propId( f ) ) );
+
+        assertNotNull( rule );
+        assertTrue( SchemaDescriptorPredicates.hasLabel( rule, labelId( LABEL1 ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( a ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( b ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( c ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( d ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( e ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( f ) ) );
+        assertEquals( NewIndexDescriptor.Type.GENERAL, rule.getIndexDescriptor().type() );
+    }
+
+    @Ignore("it is currently disputed whether constraints should be supported via Core API or not.")
+    @Test
+    public void shouldReturnConstraintRuleForLabelAndPropertyComposite() throws Exception
+    {
+        String a = "a", b = "b", c = "c", d = "d", e = "e", f = "f";
+        createSchema( db ->
+        {
+            db.schema().constraintFor( Label.label( LABEL1 ) )
+              .assertPropertyIsUnique( a ).assertPropertyIsUnique( b ).assertPropertyIsUnique( c )
+              .assertPropertyIsUnique( d ).assertPropertyIsUnique( e ).assertPropertyIsUnique( f ).create();
+        } );
+
+        IndexRule rule = storage.indexGetForSchema( NewIndexDescriptorFactory.forLabel(
+                labelId( LABEL1 ), propId( a ), propId( b ), propId( c ), propId( d ), propId( e ), propId( f ) ) );
+
+        assertNotNull( rule );
+        assertTrue( SchemaDescriptorPredicates.hasLabel( rule, labelId( LABEL1 ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( a ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( b ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( c ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( d ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( e ) ) );
+        assertTrue( SchemaDescriptorPredicates.hasProperty( rule, propId( f ) ) );
+        assertEquals( NewIndexDescriptor.Type.UNIQUE, rule.getIndexDescriptor().type() );
```

Comment:
Also you can do assert directly on `NewIndexDescriptor` equality if you don't care to see exactly what was wrong. Maybe we do in this case.